pandas/core/dtypes/concat.py

```diff
@@ -471,7 +471,16 @@ def _concat_datetimetz(to_concat, name=None):
     all inputs must be DatetimeIndex
     it is used in DatetimeIndex.append also
     """
-    return to_concat[0]._concat_same_dtype(to_concat, name=name)
+    from pandas.core.arrays.datetimelike import DatetimeLikeArrayMixin
```

Comment:
this is for all three, not just DatetimeArray?

Reply:
Sorry, I think I reverted the wrong one. This should be OK as is on master.